File: backend/models/Article.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;

const RequestSchema = new Schema({
  googleDocId: {
    type: String,
    required:true
  },
  title: {
    type: String,
    required:true
  },
  subtitle: {
    type: String,
  },
  isFeatured: {
    type: Boolean,
    required:true
  },
  image_link: {
    type: String,
    required: false 
  },
  author: {
    type: String,
    required: true
  },
  genre: {
    type: String,
    required: true,
  },
  content: {
    type: String,
    required: true 
  },
  clicks: {
    type: Number,
    required: true,
    default: 0
  },
  date: {
    type: Date,
    default: Date.now
  },
});

module.exports = mongoose.model("article", RequestSchema);

File: gazebo/lib/employee/utilities/Article.dart
class Article {
  String? Title;
  String? Subtitle;
  String? Genre;
  String? isFeatured;
  String? Author;
  Article(
      {title = null,
      subtitle = null,
      genre = null,
      featured = null,
      author = null}) {
    this.Title = title;
    this.Subtitle = subtitle;
    this.Genre = genre;
    this.isFeatured = featured;
    this.Author = author;
  }
  void setTitle(title) {
    this.Title = title;
  }

  void setSubtitle(subtitle) {
    this.Subtitle = subtitle;
  }

  void setGenre(genre) {
    this.Genre = genre;
  }

  void setIsFeatured(featured) {
    this.isFeatured = featured;
  }

  void setAuthor(author) {
    this.Author = author;
  }

  Map<String, dynamic> getAsMap() {
    return {
      "title": this.Title,
      "subtitle": this.Subtitle,
      "genre": this.Genre,
      "isFeatured": this.isFeatured,
      "author": this.Author
    };
  }
}
